huaweicloud/resource_huaweicloud_scm_certificate_v3.go

```diff
@@ -0,0 +1,353 @@
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"authentifications": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"record_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"record_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"record_value": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"domain": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					}},
+			},
+		},
+	}
+}
+
+func resourceScmCertificateV3Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	scmClient, err := config.ScmV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud SCM client: %s", err)
+	}
+
+	importOpts := certificates.ImportOpts{
+		Name:             d.Get("name").(string),
+		PrivateKey:       d.Get("private_key").(string),
+		Certificate:      d.Get("certificate").(string),
+		CertificateChain: d.Get("certificate_chain").(string),
+	}
+
+	c, err := certificates.Import(scmClient, importOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error importing certificate: %s", err)
+	}
+	// If all has been successful, set the ID on the resource
+	d.SetId(c.CertificateId)
+	log.Printf("[DEBUG] Imported certificate %s: %#v", d.Id(), importOpts)
+
+	pushCert := d.Get("target").([]interface{})
+	for _, pushInfo := range pushCert {
+		targetService := pushInfo.(map[string]interface{})["service"].(string)
+		targetProjectArr := pushInfo.(map[string]interface{})["project"].([]interface{})
+
+		for _, targetProject := range targetProjectArr {
+			pushOpts := certificates.PushOpts{
+				TargetProject: targetProject.(string),
+				TargetService: targetService,
+			}
+			log.Printf("[DEBUG] Push certificate to services. %#v", pushOpts)
+			err := doPushCertificateToService(d.Id(), pushOpts, scmClient)
+			if err != nil {
+				d.SetId("")
+				return err
+			}
+		}
+	}
+
+	return resourceScmCertificateV3Read(d, meta)
+}
+
+func resourceScmCertificateV3Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	scmClient, err := config.ScmV3Client(GetRegion(d, config))
+
+	oldVal, newVal := d.GetChange("target")
+	newPushCert, err := parsePushCertificateToMap(newVal.([]interface{}))
+	if err != nil {
+		return err
+	}
+	oldPushCert, _ := parsePushCertificateToMap(oldVal.([]interface{}))
+
+	// extract the new push service
+	for service, newProjects := range newPushCert {
+		oldProjects, ok := oldPushCert[service]
+		if strings.Compare(service, TARGET_SERVICE_CDN) == 0 && !ok {
+			pushOpts := certificates.PushOpts{
+				TargetService: service,
+			}
+			log.Printf("[DEBUG] Find new services and start to push. %#v", pushOpts)
+			err := doPushCertificateToService(d.Id(), pushOpts, scmClient)
+			if err != nil {
+				d.SetId("")
```

Comment:
we can not set the ID to null in Update method, just return the error.

Reply:
If id is not set to blank, the data will be saved to the state file. The change cannot be recognized when executed again.
To be safe, I overwrite the `target` with the old value.